### PR TITLE
Add param to prevent user creation

### PR DESCRIPTION
**Pull request by b8ne**

### CODE REVIEW

**Comment by Jono20201** [comment]
I don't think this is a good solution? The decision for this should be stored in configuration files rather than allowing anyone (potentially) to decide if they want to register or not by adding/removing a POST parameter.

File: src/FacebookLoginTrait.php [hunk 1]
```diff
@@ -56,7 +52,14 @@ public function loginFacebook(Request $request)
 
                 $user = $userModel::where($facebook_id_column, $fbUser['id'])->first();
 
+                /**
+                 * Create a new user if they haven't already signed up.
+                 */
                 if (!$user) {
+                    if ($request->has('prevent_user_create') && $request->get('prevent_user_create')) {
```
Comment (Jono20201) [inline src/FacebookLoginTrait.php:59]: Store the option for this in the configuration file instead?

File: src/FacebookLoginTrait.php [hunk 2]
```diff
@@ -43,10 +43,6 @@ public function loginFacebook(Request $request)
                  * Check if the user has already signed up.
                  */
                 $userModel = config('auth.providers.users.model');
-
-                /**
-                 * Create a new user if they haven't already signed up.
-                 */
                 $facebook_id_column = config('facebook.registration.facebook_id', 'facebook_id');
                 $name_column        = config('facebook.registration.name', 'name');
```
Comment (Jono20201) [inline src/FacebookLoginTrait.php:47]: If we are not creating a user we don't need to assign these variables, maybe move them down? (We only need to know the facebook id column.

File: src/FacebookLoginTrait.php [hunk 3]
```diff
@@ -80,6 +83,8 @@ public function loginFacebook(Request $request)
                     if (!is_null(config('facebook.registration.attach_role'))) {
                         $user->attachRole(config('facebook.registration.attach_role'));
                     }
+
+                    return $user;
```
Comment (Jono20201) [inline src/FacebookLoginTrait.php:87]: If this line needed?

**Review by Jono20201 — CHANGES_REQUESTED** [review]

**Comment by danjdewhurst** [comment]
As @Jono20201 has said, this should be an option given in the configuration files, not a parameter in the request.

As for for returning the user model, I don't believe it's needed.

**Pull request closed without merging**